### ISSUE TITLE
Bug: Some values don't refresh. KeyError: DecimalDigits

### ISSUE THREAD

**Issue by michaelf988** [johnaherninfotrack/homeassistant_custom_siemensozw672#9]
With version 0.2.5 some values don't refresh. Debug throws some errors.
There must be some issues with numeric values.
I have to downgrade to version 0.2.1.

```
2024-01-05 19:14:58.200 WARNING (MainThread) [custom_components.siemens_ozw672] OZW672 Data Poll time exceeding 60 seconds. Last Poll Time: 71 seconds
2024-01-05 19:14:58.234 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 243, in _handle_refresh_interval
    await self._async_refresh(log_failures=True, scheduled=True)
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 399, in _async_refresh
    self.async_update_listeners()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 182, in async_update_listeners
    update_callback()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 479, in _handle_coordinator_update
    self.async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 745, in async_write_ha_state
    self._async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 845, in _async_write_ha_state
    state, attr = self._async_generate_attributes()
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 786, in _async_generate_attributes
    state = self._stringify_state(available)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 751, in _stringify_state
    if (state := self.state) is None:
                 ^^^^^^^^^^
  File "/config/custom_components/siemens_ozw672/sensor.py", line 312, in state
    if data.isnumeric() and self.config_entry["DPDescr"]["DecimalDigits"] != "0":
                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
KeyError: 'DecimalDigits'
```

```
2024-01-06 09:08:59.434 DEBUG (MainThread) [custom_components.siemens_ozw672] API GET: {'Data': {'Type': 'Numeric', 'Value': '       33.4', 'Unit': '°C'}, 'Result': {'Success': 'true'}}
2024-01-06 09:08:59.435 DEBUG (MainThread) [custom_components.siemens_ozw672] async_get_data response : {'Data': {'Type': 'Numeric', 'Value': '       33.4', 'Unit': '°C'}, 'Result': {'Success': 'true'}}
2024-01-06 09:08:59.435 DEBUG (MainThread) [custom_components.siemens_ozw672] HTTP GET url: http://192.168.1.52/api/menutree/read_datapoint.json?SessionId=XXXXXX&Id=1172
2024-01-06 09:08:59.746 DEBUG (MainThread) [custom_components.siemens_ozw672] API GET: {'Data': {'Type': 'RadioButton', 'Value': 'Aus', 'Unit': ''}, 'Result': {'Success': 'true'}}
2024-01-06 09:08:59.746 DEBUG (MainThread) [custom_components.siemens_ozw672] async_get_data response : {'Data': {'Type': 'RadioButton', 'Value': 'Aus', 'Unit': ''}, 'Result': {'Success': 'true'}}
2024-01-06 09:08:59.746 DEBUG (MainThread) [custom_components.siemens_ozw672] HTTP GET url: http://192.168.1.52/api/menutree/read_datapoint.json?SessionId=XXXXXX&Id=1179
...
2024-01-06 09:09:08.967 DEBUG (MainThread) [custom_components.siemens_ozw672] async_get_data response : {'Data': {'Type': 'RadioButton', 'Value': 'Nein', 'Unit': ''}, 'Result': {'Success': 'true'}}
2024-01-06 09:09:08.967 DEBUG (MainThread) [custom_components.siemens_ozw672] OZW672 Data Poll time: 49 seconds
2024-01-06 09:09:08.967 DEBUG (MainThread) [custom_components.siemens_ozw672] SiemensOzw672BinarySwitch: Config: {'Id': '559', 'WriteAccess': 'true', 'OpLine': '479', 'Name': 'Heizkennlinie Adaption Heizkreis 1', 'MenuItem': 'Heizkreis 1', 'DPDescr': {'Type': 'RadioButton', 'Name': 'Heizkennlinie Adaption Heizkreis 1', 'Buttons': [{'TextOpt0': 'Aus', 'TextOpt1': 'Ein', 'Significance': '1', 'IsActive': 'false'}], 'HAType': 'switch'}, 'entry_id': 'e9581dd998f7c2c10f6e448d4383fbb5_479', 'device_id': 'e9581dd998f7c2c10f6e448d4383fbb5', 'device_name': 'LMS15.003A100', 'entity_prefix': '479 '}
...
2024-01-06 09:09:08.978 DEBUG (MainThread) [custom_components.siemens_ozw672] SiemensOzw672TempControl: Data: {'553': {'Data': {'Type': 'Enumeration', 'Value': 'Automatik', 'Unit': '', 'EnumValue': '1'}, 'Result': {'Success': 'true'}}, '554': {'Data': {'Type': 'Numeric', 'Value': '       22.0', 'Unit': '°C'}, 'Result': {'Success': 'true'}}, '555': {'Data': {'Type': 'Numeric', 'Value': '       18.0', 'Unit': '°C'}, 'Result': {'Success': 'true'}}, '556': {'Data': {'Type': 'Numeric', 'Value': '        5.5', 'Unit': '°C'}, 'Result': {'Success': 'true'}}, '557': {'Data': {'Type': 'Numeric', 'Value': '       0.88', 'Unit': ''}, 'Result': {'Success': 'true'}}, '559': {'Data': {'Type': 'RadioButton', 'Value': 'Aus', 'Unit': ''}, 'Result': {'Success': 'true'}}, '558': {'Data': {'Type': 'Numeric', 'Value': '        0.0', 'Unit': '°C'}, 'Result': {'Success': 'true'}}, '560': {'Data': {'Type': 'Numeric', 'Value': '       15.0', 'Unit': '°C'}, 'Result': {'Success': 'true'}}, '561': {'Data': {'Type': 'Numeric', 'Value': '         0', 'Unit': '°C'}, 'Result': {'Success': 'true'}}, '566': {'Data': {'Type': 'Numeric', 'Value': '       300', 'Unit': 's'}, 'Result': {'Success': 'true'}}, '591': {'Data': {'Type': 'Enumeration', 'Value': 'Komfort', 'Unit': '', 'EnumValue': '2'}, 'Result': {'Success': 'true'}}, '592': {'Data': {'Type': 'Enumeration', 'Value': 'Automatik', 'Unit': '', 'EnumValue': '4'}, 'Result': {'Success': 'true'}}, '594': {'Data': {'Type': 'Enumeration', 'Value': 'Automatik', 'Unit': '', 'EnumValue': '1'}, 'Result': {'Success': 'true'}}, '595': {'Data': {'Type': 'Numeric', 'Value': '       21.0', 'Unit': '°C'}, 'Result': {'Success': 'true'}}, '596': {'Data': {'Type': 'Numeric', 'Value': '       16.0', 'Unit': '°C'}, 'Result': {'Success': 'true'}}, '597': {'Data': {'Type': 'Numeric', 'Value': '       10.0', 'Unit': '°C'}, 'Result': {'Success': 'true'}}, '598': {'Data': {'Type': 'Numeric', 'Value': '       0.70', 'Unit': ''}, 'Result': {'Success': 'true'}}, '599': {'Data': {'Type': 'Numeric', 'Value': '        0.0', 'Unit': '°C'}, 'Result': {'Success': 'true'}}, '600': {'Data': {'Type': 'RadioButton', 'Value': 'Aus', 'Unit': ''}, 'Result': {'Success': 'true'}}, '601': {'Data': {'Type': 'Numeric', 'Value': '       15.0', 'Unit': '°C'}, 'Result': {'Success': 'true'}}, '602': {'Data': {'Type': 'Numeric', 'Value': '        -3', 'Unit': '°C'}, 'Result': {'Success': 'true'}}, '604': {'Data': {'Type': 'Numeric', 'Value': '        38', 'Unit': '°C'}, 'Result': {'Success': 'true'}}, '607': {'Data': {'Type': 'Numeric', 'Value': '       180', 'Unit': 's'}, 'Result': {'Success': 'true'}}, '632': {'Data': {'Type': 'Enumeration', 'Value': 'Komfort', 'Unit': '', 'EnumValue': '2'}, 'Result': {'Success': 'true'}}, '633': {'Data': {'Type': 'Enumeration', 'Value': 'Automatik', 'Unit': '', 'EnumValue': '4'}, 'Result': {'Success': 'true'}}, '635': {'Data': {'Type': 'Enumeration', 'Value': 'Ein', 'Unit': '', 'EnumValue': '1'}, 'Result': {'Success': 'true'}}, '636': {'Data': {'Type': 'Numeric', 'Value': '        57', 'Unit': '°C'}, 'Result': {'Success': 'true'}}, '637': {'Data': {'Type': 'Numeric', 'Value': '         8', 'Unit': '°C'}, 'Result': {'Success': 'true'}}, '638': {'Data': {'Type': 'Numeric', 'Value': '        65', 'Unit': '°C'}, 'Result': {'Success': 'true'}}, '639': {'Data': {'Type': 'Enumeration', 'Value': 'Zeitprogramm 4', 'Unit': '', 'EnumValue': '2'}, 'Result': {'Success': 'true'}}, '648': {'Data': {'Type': 'Enumeration', 'Value': 'Trinkwasser Freigabe', 'Unit': '', 'EnumValue': '2'}, 'Result': {'Success': 'true'}}, '649': {'Data': {'Type': 'Enumeration', 'Value': 'Ein', 'Unit': '', 'EnumValue': '1'}, 'Result': {'Success': 'true'}}, '651': {'Data': {'Type': 'Enumeration', 'Value': 'Aus', 'Unit': '', 'EnumValue': '1'}, 'Result': {'Success': 'true'}}, '658': {'Data': {'Type': 'Numeric', 'Value': '         2', 'Unit': 'min'}, 'Result': {'Success': 'true'}}, '659': {'Data': {'Type': 'Numeric', 'Value': '        10', 'Unit': 'min'}, 'Result': {'Success': 'true'}}, '660': {'Data': {'Type': 'Numeric', 'Value': '        20', 'Unit': '°C'}, 'Result': {'Success': 'true'}}, '681': {'Data': {'Type': 'Numeric', 'Value': '        6.0', 'Unit': '°C'}, 'Result': {'Success': 'true'}}, '682': {'Data': {'Type': 'Numeric', 'Value': '        6.0', 'Unit': '°C'}, 'Result': {'Success': 'true'}}, '683': {'Data': {'Type': 'Numeric', 'Value': '       12.0', 'Unit': '°C'}, 'Result': {'Success': 'true'}}, '684': {'Data': {'Type': 'Numeric', 'Value': '        7.0', 'Unit': '°C'}, 'Result': {'Success': 'true'}}, '685': {'Data': {'Type': 'Numeric', 'Value': '        6.0', 'Unit': '°C'}, 'Result': {'Success': 'true'}}, '686': {'Data': {'Type': 'Numeric', 'Value': '        8.0', 'Unit': '°C'}, 'Result': {'Success': 'true'}}, '760': {'Data': {'Type': 'RadioButton', 'Value': 'Ein', 'Unit': ''}, 'Result': {'Success': 'true'}}, '749': {'Data': {'Type': 'Numeric', 'Value': '        18', 'Unit': '°C'}, 'Result': {'Success': 'true'}}, '752': {'Data': {'Type': 'Numeric', 'Value': '       120', 'Unit': 'min'}, 'Result': {'Success': 'true'}}, '782': {'Data': {'Type': 'RadioButton', 'Value': 'Ein', 'Unit': ''}, 'Result': {'Success': 'true'}}, '783': {'Data': {'Type': 'RadioButton', 'Value': 'Ein', 'Unit': ''}, 'Result': {'Success': 'true'}}, '819': {'Data': {'Type': 'Numeric', 'Value': '        34', 'Unit': 'h'}, 'Result': {'Success': 'true'}}, '1077': {'Data': {'Type': 'Enumeration', 'Value': 'Heizbetrieb Komfort', 'Unit': '', 'EnumValue': '114'}, 'Result': {'Success': 'true'}}, '1078': {'Data': {'Type': 'Enumeration', 'Value': 'Heizbetrieb Komfort', 'Unit': '', 'EnumValue': '114'}, 'Result': {'Success': 'true'}}, '1080': {'Data': {'Type': 'Enumeration', 'Value': 'Geladen, Nenntemperatur', 'Unit': '', 'EnumValue': '99'}, 'Result': {'Success': 'true'}}, '1081': {'Data': {'Type': 'Enumeration', 'Value': 'In Teillastbetrieb für HK', 'Unit': '', 'EnumValue': '167'}, 'Result': {'Success': 'true'}}, '1084': {'Data': {'Type': 'Enumeration', 'Value': 'In Betrieb', 'Unit': '', 'EnumValue': '18'}, 'Result': {'Success': 'true'}}, '1091': {'Data': {'Type': 'Numeric', 'Value': '       40.6', 'Unit': '°C'}, 'Result': {'Success': 'true'}}, '1092': {'Data': {'Type': 'Numeric', 'Value': '0', 'Unit': '°C'}, 'Result': {'Success': 'true'}}, '1093': {'Data': {'Type': 'Numeric', 'Value': '       38.6', 'Unit': '°C'}, 'Result': {'Success': 'true'}}, '1095': {'Data': {'Type': 'Numeric', 'Value': '       44.6', 'Unit': '°C'}, 'Result': {'Success': 'true'}}, '1097': {'Data': {'Type': 'Numeric', 'Value': '       36.1', 'Unit': '°C'}, 'Result': {'Success': 'true'}}, '1101': {'Data': {'Type': 'Numeric', 'Value': '         0', 'Unit': '%'}, 'Result': {'Success': 'true'}}, '1103': {'Data': {'Type': 'TimeOfDay', 'Value': '4129', 'Unit': 'h'}, 'Result': {'Success': 'true'}}, '1104': {'Data': {'Type': 'Numeric', 'Value': '      8742', 'Unit': ''}, 'Result': {'Success': 'true'}}, '1105': {'Data': {'Type': 'TimeOfDay', 'Value': '3550', 'Unit': 'h'}, 'Result': {'Success': 'true'}}, '1106': {'Data': {'Type': 'TimeOfDay', 'Value': '644', 'Unit': 'h'}, 'Result': {'Success': 'true'}}, '1143': {'Data': {'Type': 'Numeric', 'Value': '        7.9', 'Unit': '°C'}, 'Result': {'Success': 'true'}}, '1144': {'Data': {'Type': 'Numeric', 'Value': '        6.3', 'Unit': '°C'}, 'Result': {'Success': 'true'}}, '1145': {'Data': {'Type': 'RadioButton', 'Value': 'Ein', 'Unit': ''}, 'Result': {'Success': 'true'}}, '1151': {'Data': {'Type': 'Numeric', 'Value': '0', 'Unit': '°C'}, 'Result': {'Success': 'true'}}, '1152': {'Data': {'Type': 'Numeric', 'Value': '       38.6', 'Unit': '°C'}, 'Result': {'Success': 'true'}}, '1154': {'Data': {'Type': 'RadioButton', 'Value': 'Ein', 'Unit': ''}, 'Result': {'Success': 'true'}}, '1160': {'Data': {'Type': 'Numeric', 'Value': '       32.5', 'Unit': '°C'}, 'Result': {'Success': 'true'}}, '1161': {'Data': {'Type': 'Numeric', 'Value': '       33.4', 'Unit': '°C'}, 'Result': {'Success': 'true'}}, '1172': {'Data': {'Type': 'RadioButton', 'Value': 'Aus', 'Unit': ''}, 'Result': {'Success': 'true'}}, '1179': {'Data': {'Type': 'Numeric', 'Value': '0', 'Unit': '°C'}, 'Result': {'Success': 'true'}}, '1175': {'Data': {'Type': 'Numeric', 'Value': '       55.0', 'Unit': '°C'}, 'Result': {'Success': 'true'}}, '1195': {'Data': {'Type': 'Numeric', 'Value': '        1.4', 'Unit': 'bar'}, 'Result': {'Success': 'true'}}, '1236': {'Data': {'Type': 'Enumeration', 'Value': '0:Kein Fehler', 'Unit': '', 'EnumValue': '0'}, 'Result': {'Success': 'true'}}, '1237': {'Data': {'Type': 'Enumeration', 'Value': 'S0/G1', 'Unit': '', 'EnumValue': '0'}, 'Result': {'Success': 'true'}}, '1238': {'Data': {'Type': 'Enumeration', 'Value': '0:Kein Fehler', 'Unit': '', 'EnumValue': '0'}, 'Result': {'Success': 'true'}}, '1239': {'Data': {'Type': 'Enumeration', 'Value': 'S0/G1', 'Unit': '', 'EnumValue': '0'}, 'Result': {'Success': 'true'}}, '1240': {'Data': {'Type': 'Enumeration', 'Value': '0:Keine Wartung anstehend', 'Unit': '', 'EnumValue': '0'}, 'Result': {'Success': 'true'}}, '1250': {'Data': {'Type': 'Numeric', 'Value': '        5.6', 'Unit': '°C'}, 'Result': {'Success': 'true'}}, '1251': {'Data': {'Type': 'Numeric', 'Value': '      -10.0', 'Unit': '°C'}, 'Result': {'Success': 'true'}}, '1252': {'Data': {'Type': 'Numeric', 'Value': '       34.9', 'Unit': '°C'}, 'Result': {'Success': 'true'}}, '870': {'Data': {'Type': 'Numeric', 'Value': '0', 'Unit': 'min'}, 'Result': {'Success': 'true'}}, '871': {'Data': {'Type': 'Numeric', 'Value': '0', 'Unit': 'h'}, 'Result': {'Success': 'true'}}, '864': {'Data': {'Type': 'Numeric', 'Value': '0', 'Unit': ''}, 'Result': {'Success': 'true'}}, '865': {'Data': {'Type': 'Numeric', 'Value': '0', 'Unit': ''}, 'Result': {'Success': 'true'}}, '872': {'Data': {'Type': 'DateTime', 'Value': '01. Oktober 2023 09:12', 'Unit': ''}, 'Result': {'Success': 'true'}}, '873': {'Data': {'Type': 'Enumeration', 'Value': '323:Wasserdruck 3 niedrig', 'Unit': '', 'EnumValue': '323'}, 'Result': {'Success': 'true'}}, '874': {'Data': {'Type': 'Numeric', 'Value': '566', 'Unit': ''}, 'Result': {'Success': 'true'}}, '875': {'Data': {'Type': 'Numeric', 'Value': '4', 'Unit': ''}, 'Result': {'Success': 'true'}}, '867': {'Data': {'Type': 'Numeric', 'Value': '0', 'Unit': 'min'}, 'Result': {'Success': 'true'}}, '868': {'Data': {'Type': 'Numeric', 'Value': '0', 'Unit': 'min'}, 'Result': {'Success': 'true'}}, '866': {'Data': {'Type': 'RadioButton', 'Value': 'Nein', 'Unit': ''}, 'Result': {'Success': 'true'}}}
2024-01-06 09:09:08.978 DEBUG (MainThread) [custom_components.siemens_ozw672] SiemensOzw672TempControl: Config: {'Id': '602', 'WriteAccess': 'true', 'OpLine': '3828', 'Name': 'Tagesheizgrenze HK2', 'MenuItem': 'Heizkreis 2', 'DPDescr': {'Type': 'Numeric', 'Value': '-3.000000', 'Unit': '°C', 'Name': 'Tagesheizgrenze HK2', 'Min': '-10.000000', 'Max': '10.000000', 'Resolution': '1.000000', 'FieldWitdh': '11', 'DecimalDigits': '0', 'HasValid': 'true', 'IsValid': 'true', 'HAType': 'number'}, 'entry_id': 'e9581dd998f7c2c10f6e448d4383fbb5_3828', 'device_id': 'e9581dd998f7c2c10f6e448d4383fbb5', 'device_name': 'LMS15.003A100', 'entity_prefix': '3828 '}
...
2024-01-06 09:09:09.004 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 243, in _handle_refresh_interval
    await self._async_refresh(log_failures=True, scheduled=True)
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 399, in _async_refresh
    self.async_update_listeners()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 182, in async_update_listeners
    update_callback()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 479, in _handle_coordinator_update
    self.async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 745, in async_write_ha_state
    self._async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 845, in _async_write_ha_state
    state, attr = self._async_generate_attributes()
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 786, in _async_generate_attributes
    state = self._stringify_state(available)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 751, in _stringify_state
    if (state := self.state) is None:
                 ^^^^^^^^^^
  File "/config/custom_components/siemens_ozw672/sensor.py", line 312, in state
    if data.isnumeric() and self.config_entry["DPDescr"]["DecimalDigits"] != "0":
                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
KeyError: 'DecimalDigits'
```

**Comment by johnaherninfotrack** [comment]
Thanks - investigating.

**Comment by suuronenh** [comment]
How do you downgrade to a previous version? The redownload dialog shows the previous versions but reverts to latest after selection. No error message is displayed. I'm experiencing this same issue. I'm also seeing some values updating rarely while others update frequently. The ones updating frequently are the ones I added first. I have a total of 38 entities added.

Thanx for this integration, awesome to be able to get the data with a few clicks!

**Comment by michaelf988** [comment]
I uninstalled it completely from HACS and reinstalled it selecting version 0.2.1.

**Comment by suuronenh** [comment]
> I uninstalled it completely from HACS and reinstalled it selecting version 0.2.1.

Thanks for answering!

Even after completely removing it from HACS I could not select any other version than 0.2.5 to be downloaded through the GUI. In fact on fresh install there is no version selection available, it always downloads the latest. Version selection is available from the re-download function but then it is stuck on 0.2.5.

I was able to install older version by manually downloading the old version zip file from github and extracting the custom_components content s to /config/custom_components/siemens_ozw672/. Then I could add the integration normally. I downloaded 0.2.4 version, I'm still testing whether the problem occurs with it too.
